icepyx/core/types.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import Literal, TypedDict
+from typing_extensions import NotRequired
+
+
+CMRParamsBase = TypedDict(
```

Comment:
I'm shooting for annotating the code as-is to enable deeper future refactors with less mistakes. I would prefer to pass around dataclasses :)

Here we're using the "old" way of defining a TypedDict because the keys we want to use are not valid in the new way. The new way is to define them as a class.